reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -43,5 +43,193 @@ promise_test(t => {
     assert_not_equals(writer.ready, readyPromise, 'the ready promise property should change');
     return promise_rejects(t, new TypeError(), writer.ready, 'the ready promise should reject with a TypeError');
   });
-
 }, 'Aborting a WritableStream should cause the writer\'s fulfilled ready promise to reset to a rejected one');
+
+promise_test(t => {
+  const ws = new WritableStream({});
+
+  const writer = ws.getWriter();
+  writer.releaseLock();
+
+  return promise_rejects(t, new TypeError(), writer.abort(), 'abort() should reject with a TypeError');
+}, 'abort() on a released writer rejects');
+
+promise_test(() => {
+  const ws = recordingWritableStream();
+
+  return delay(0)
+    .then(() => {
+      const writer = ws.getWriter();
+
+      writer.abort();
+      writer.write(1);
+      writer.write(2);
+    })
+    .then(() => {
+      assert_equals(ws.events.length, 2, 'the stream should have received 2 events');
+      assert_equals(ws.events[0], 'abort', 'abort should be called first');
+      assert_equals(ws.events[1], undefined, 'no further writes should have happened');
+    });
+}, 'Aborting a WritableStream immediately prevents future writes');
+
+promise_test(() => {
+  const ws = recordingWritableStream();
+
+  return delay(0)
+    .then(() => {
+      const writer = ws.getWriter();
+
+      writer.write(1);
+      writer.write(2);
+      writer.write(3);
+      writer.abort();
+      writer.write(4);
+      writer.write(5);
+    }).then(() => {
+      assert_equals(ws.events.length, 4, 'the stream should have received 4 events');
+      assert_equals(ws.events[1], 1, 'chunk should be 1');
+      assert_equals(ws.events[2], 'abort', 'abort should be called after write');
+      assert_equals(ws.events[3], undefined, 'no further writes should have happened');
+    });
+}, 'Aborting a WritableStream prevents further writes after any that are in progress');
+
+promise_test(() => {
+  const ws = new WritableStream({
+    abort() {
+      return 'Hello';
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  const abortPromise = writer.abort('a');
+  return abortPromise.then(value => {
+    assert_equals(value, undefined, 'fulfillment value must be undefined');
+  });
+}, 'Fulfillment value of ws.abort() call must be undefined even if the underlying sink returns a non-undefined value');
+
+promise_test(t => {
+  const errorInSinkAbort = new Error('Sorry, it just wasn\'t meant to be.');
+  const ws = new WritableStream({
+    abort() {
+      throw errorInSinkAbort;
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  return promise_rejects(t, errorInSinkAbort, writer.abort(undefined),
```

Comment:
Can you use the `error1` pattern used elsewhere? With a single `error1` at the top of the file? Here and below.

Also everywhere that creates a separate `passedReason` can use that `error1`.